src/tools/run-make-support/src/run.rs

```diff
@@ -21,6 +21,20 @@ fn run_common(name: &str, args: Option<&[&str]>) -> Command {
         // will have to be changed (and the support files will have to be uploaded).
         cmd.arg("0");
         cmd.arg(bin_path);
+        cmd
+    } else if let Ok(runner) = std::env::var("RUNNER") {
+        let mut args = split_maybe_args(&runner);
+
+        let prog = args.remove(0);
+        let mut cmd = Command::new(prog);
+
+        for arg in args {
+            cmd.arg(arg);
+        }
+
+        cmd.arg("--");
+        cmd.arg(bin_path);
+
         cmd
```

Comment:
So, I _think_  the problem is that the wasm tests set the (bootstrap) `runner` (probably to `wasmtime`), that now gets forwarded to `run-make` where it didn't before, and then this bit of code tries to execute any `run` command with that runner. But the tests in question don't actually expect that, they run on the host instead.

I'm not sure about this, but I think that means that these tests were never really testing what they were supposed to anyway? E.g. `alloc-no-oom-handling` appears to, currently, just ignore the target completely, and compile and run a host binary.

Reply:
Hm yeah, e.g. `alloc-no-oom-handling` are broken re. cross-compile, because it's not cross-by-default, but only opt-in cross ATM with sth like `rustc().target(())`. I'll try to get https://github.com/rust-lang/rust/pull/139244 working next week, which might make your life here easier.